Add script for generating `MuJoCo/Ant-v5`

Kallinteris-Andreas opened the pull request:
I have an expert policy (6.7k return) for the `MuJoco/Ant-v5` environment (previous best was 5.8k from what I can tell)

I would like a review of the `create_dataset.py` before I run it, any other comments are welcome

younik commented on scripts/ant/create_dataset.py:
```diff
@@ -0,0 +1,78 @@
+from minari import DataCollectorV0, StepDataCallback
+from stable_baselines3 import A2C, PPO, SAC, TD3
+import stable_baselines3
+assert stable-baselines3.__version__ == "2.0.0a5"
```
what is this line for?

younik reviewed (commented):
This sounds good to me; we may want to use a different dataset name maybe

Kallinteris-Andreas commented:
What dataset name should i use

younik commented:
> What dataset name should i use

uhm, maybe `ant-v5-expert-v0`?

Kallinteris-Andreas commented:
I have run the script, and it has generated 49 episodes with `200k` steps plus 4 episodes with 10k+ steps each (totaling `200k` steps)

Is that the desired behavior, or should I set the `max_episode_length` to `1000`?

I am asking since minari has a `dataset.sample_episodes()` and not a `dataset.sample_steps()`

younik commented on scripts/ant/create_dataset.py:
```diff
@@ -0,0 +1,78 @@
+
+    obs, rew, terminated, truncated, info = collector_env.step(action)
+    # Checkpoint
+    if (n_step + 1) % 200e3 == 0:
+        print(f"STEPS RECORDED: {n_step}")
+        if dataset is None:
+            dataset = minari.create_dataset_from_collector_env(
+                collector_env=collector_env,
+                dataset_id=dataset_name,
+                algorithm_name="SB3/SAC",
+                code_permalink="https://github.com/Kallinteris-Andreas/gymnasium-mujuco-v5-envs-validation/blob/main/create_dataset.py",
+                author="Kallinteris Andreas",
+                author_email="kallinteris@protonmail.com",
+            )
+        dataset.update_dataset_from_collector_env(collector_env)
+
+    if terminated or truncated:
+        env.reset()
+
+# dataset.update_dataset_from_collector_env(collector_env)
+collector_env.save_to_disk("test.hdf5")
```
This is not the right workflow: you want to use `create_dataset_from_collector_env` at the end, outside the for loop (instead of `collector_env.save_to_disk("test.hdf5")`)

No need to use `dataset.update_dataset_from_collector_env(collector_env)`then


You can see the sample workflow here: https://minari.farama.org/content/basic_usage/#save-dataset

younik reviewed (changes requested):
Oh sorry, I missed the last part of the workflow, that may cause your problem. See the review comment

Kallinteris-Andreas commented:
Ok, I have updated it, recheck the script (Note: I will clean it up after you have confirmed the functionality is ok)

younik commented on scripts/ant/create_dataset.py:
```diff
@@ -0,0 +1,78 @@
+    author="Kallinteris Andreas",
+    author_email="kallinteris@protonmail.com",
+)
+collector_env.save_to_disk("test.hdf5")
```
remove this line, then LGTM

younik commented on scripts/ant/create_dataset.py:
```diff
@@ -0,0 +1,78 @@
+for n_step in range(NUM_STEPS):
+    action, _ = model.predict(obs, deterministic=True)
+    # Add some noise to each step action
+    action += np.random.randn(*action.shape) * POLICY_NOISE
```
why do we want to have noise on the policy actions?

Kallinteris-Andreas commented:
Update:
I have 
- renamed the dataset to `"ant-v5-expert-tuned-v0"`
- reduced step count down to 2mil (to be consistent with other datasets)
-  set `record_info=False` because https://github.com/Farama-Foundation/Minari/issues/191
- updated to `minari==0.4.3`


Note: I get these warnings, are any of them a problem?
```py
/home/master-andreas/gym/rl/project/temp_env/lib/python3.11/site-packages/minari/utils.py:394: UserWarning: `minari_version` is set to None. The compatible dataset version specifier for Minari will be set to ~=0.4.
  warnings.warn(
/home/master-andreas/gym/rl/project/temp_env/lib/python3.11/site-packages/minari/utils.py:414: UserWarning: `eval_env` is set to None. If another environment is intended to be used for evaluation please specify corresponding Gymnasium environment (gym.Env | gym.envs.registration.EnvSpec).              If None the environment used to collect the data (`env=None`) will be used for this purpose.
  warnings.warn(
/home/master-andreas/gym/rl/project/temp_env/lib/python3.11/site-packages/minari/utils.py:435: UserWarning: env_spec is None, no environment spec is provided during collection for this dataset
  warnings.warn(
```
@younik

younik commented on scripts/ant/create_dataset.py:
```diff
@@ -0,0 +1,72 @@
+from minari import DataCollector, StepDataCallback
+from stable_baselines3 import A2C, PPO, SAC, TD3
+
+from envs import ant_v5_release
```
what is this `env`?

younik commented on scripts/ant/create_dataset.py:
```diff
@@ -0,0 +1,72 @@
+    if terminated or truncated:
+        env.reset()
+
+dataset = minari.create_dataset(
```
This will change to `collector_env.create_dataset`

younik reviewed (commented):
The code looks good, thanks!

So you are proposing to upload this dataset to Minari remote, right? 
@rodrigodelazcano what are your thoughts on that?

Kallinteris-Andreas commented:
Yes, it is my intention for it to be uploaded to the Minari dataset remote As the policy used is by far the best, It is able to achieve a return of 6.7k While the previous best is below 6k

Kallinteris-Andreas commented:
@younik is there anything blocking it merge?

younik commented:
> @younik is there anything blocking it merge?

I belline the -v5 environments are available in gymnasium 1.0, isn't it? We should wait for that release

Kallinteris-Andreas commented:
```py
MinariDatasetSpec(env_spec=EnvSpec(id='Ant-v5', entry_point='gymnasium.envs.mujoco.ant_v5:AntEnv', reward_threshold=6000.0, nondeterministic=False, max_episode_steps=1000, order_enforce=True, disable_env_checker=False, kwargs={}, namespace=None, name='Ant', version=5, additional_wrappers=(), vector_entry_point=None), total_episodes=2026, total_steps=2000000, dataset_id='mujoco/ant/expert-v0', combined_datasets=[], observation_space=Box(-inf, inf, (105,), float64), action_space=Box(-1.0, 1.0, (8,), float32), data_path='/home/master-andreas/.minari/datasets/mujoco/ant/expert-v0/data', minari_version='0.5.1', namespace='mujoco/ant', dataset_name='expert', version=0)
```

Final version (hopefully), spec is fine, and it is reproducable

Kallinteris-Andreas commented:
i can not merge
